packages/yourloops/pages/profile/profile-page-context.tsx

```diff
@@ -0,0 +1,153 @@
+    lastName: user.lastName,
+    units: user.settings?.units?.bg ?? Units.gram,
+    birthday: user.birthday,
+    birthPlace: user.profile?.patient?.birthPlace ?? '',
+    lang: user.preferences?.displayLanguageCode ?? getCurrentLang(),
+    hcpProfession: user.profile?.hcpProfession ?? HcpProfession.empty,
+    referringDoctor: user.profile?.patient?.referringDoctor ?? undefined,
+    feedbackAccepted: !!user?.profile?.contactConsent?.isAccepted,
+    sex: user.profile?.patient?.sex ?? undefined,
+    ins: user.profile?.patient?.ins ?? undefined,
+    ssn: user.profile?.patient?.ssn ?? undefined,
+    birthFirstName: user.profile?.patient?.birthFirstName ?? undefined,
+    birthLastName: user.profile?.patient?.birthLastName ?? undefined,
+    birthNames: user.profile?.patient?.birthNames ?? undefined,
+    birthPlaceInseeCode: user.profile?.patient?.birthPlaceInseeCode ?? undefined,
+    oid: user.profile?.patient?.oid ?? undefined
+  })
+  const [saving, setSaving] = useState<boolean>(false)
+
+  const errors: ProfileErrors = {
+    firstName: !profileForm.firstName,
+    lastName: !profileForm.lastName,
+    hcpProfession: user.isUserHcp() && profileForm.hcpProfession === HcpProfession.empty,
+    birthday: user.isUserPatient() && !REGEX_BIRTHDATE.test(profileForm.birthday),
+    ins: user.isUserPatient() && (!profileForm.ins || (profileForm.ins && profileForm.ins.length !== 15)),
+    ssn: user.isUserPatient() && (!profileForm.ssn || (profileForm.ssn && profileForm.ssn.length !== 15)),
+    sex: user.isUserPatient() && !profileForm.sex,
+    birthFirstName: user.isUserPatient() && !profileForm.birthFirstName,
+    birthLastName: user.isUserPatient() && !profileForm.birthLastName,
+    birthNames: user.isUserPatient() && !profileForm.birthNames,
+    birthPlaceInseeCode: user.isUserPatient() && !profileForm.birthPlaceInseeCode,
+    oid: user.isUserPatient() && !profileForm.oid
+  }
+
+  const updatedProfile = useMemo<Profile>(() => {
+    const profile: Profile = {
+      ...user.profile,
+      firstName: profileForm.firstName,
+      lastName: profileForm.lastName,
+      fullName: `${profileForm.firstName} ${profileForm.lastName}`
+    }
+
+    if (user.isUserPatient()) {
+      profile.patient = {
+        ...profile.patient,
+        birthday: profileForm.birthday,
+        birthPlace: profileForm.birthPlace,
+        ins: profileForm.ins,
+        sex: profileForm.sex,
+        ssn: profileForm.ssn,
+        referringDoctor: profileForm.referringDoctor,
+        birthFirstName: profileForm.birthFirstName,
+        birthLastName: profileForm.birthLastName,
+        birthNames: profileForm.birthNames,
+        birthPlaceInseeCode: profileForm.birthPlaceInseeCode,
+        oid: profileForm.oid
+      }
+    }
+
+    if (user.isUserHcp()) {
+      profile.hcpProfession = profileForm.hcpProfession
+    }
+
+    if (user.isUserHcp() && !!user?.profile?.contactConsent?.isAccepted !== profileForm.feedbackAccepted) {
```

Comment:
Given that `isAccepted` is a boolean and `undefined` would be falsy, I don't understand why the `!!` would be necessary

```suggestion
    if (user.isUserHcp() && user?.profile?.contactConsent?.isAccepted !== profileForm.feedbackAccepted) {
```

Reply:
nice shot, didn't know why 😅

Reply:
`profileForm.feedbackAccepted` is a boolean and `user?.profile?.contactConsent?.isAccepted` could be undefined. Here we need to compate two booleans or the test could be wrong when testing `undefined !== false`